les/odr_requests.go

```diff
@@ -164,7 +169,15 @@ func (r *HeaderRequest) Validate(db ethdb.Database, msg *Msg) error {
 		return errInvalidMessageType
 	}
 	headers := msg.Obj.([]*types.Header)
-	if len(headers) != 1 {
+	if len(headers) == 0 && r.isByHash() {
+		// For requests by number, we only send to peers for which we know the block number
+		// is within the range of what they have, so if they don't send us the header we reject
+		// the response and try other peers.
+		// However, for requests by hash, we have no way of knowing ahead of time whether the peer
+		// should have it or not (e.g. what if there is no such block?).  So we need to accept
+		// 'no match' as a valid response, to avoid ODR endlessly trying to send to different peers.
```

Comment:
What does ODR stand for?

Reply:
On-demand retrieval: https://github.com/celo-org/celo-blockchain/blob/1188111ac106a92c6c9d89c5ff3da265b404a161/light/lightchain.go#L17

Though I've also heard "on-demand requests" as the more guessable meaning.